Extension/src/LanguageServer/extension.ts

```diff
@@ -250,35 +250,41 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
                 execSync(uninstallCommand);
             }
             execSync(installCommand);
+            clearInterval(insiderUpdateTimer);
             return Promise.resolve();
         } catch (error) {
-            return Promise.reject(new Error('Install command failed'));
+            return Promise.reject(new Error('Failed to install vsix'));
         }
     });
 }
 
 async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
-    return getTargetBuildURL(updateChannel).then((downloadUrl) => {
-        if (!downloadUrl) {
-            return Promise.resolve();
-        }
-
-        // Create a temporary file, download the vsix to it, then install the vsix
-        tmp.file({postfix: '.vsix'}, async (err, vsixPath, fd, cleanupCallback) => {
-            if (err) {
-                cleanupCallback();
-                return Promise.reject(new Error('Failed to create vsix file'));
+    const p: Promise<void> = new Promise<void>((resolve, reject) => {
+        getTargetBuildUrl(updateChannel).then((downloadUrl) => {
+            if (!downloadUrl) {
+                return resolve();
```

Comment:
I don't think it is technically correct to `return resolve();`  `resolve()` is a function that returns nothing (void).  You might be confusing it with `return Promise.resolve();` which creates a new, resolved promise and returns it.

I think it is more correct to do:

```
    resolve();
    return;
```